core/modules/utils/filesystem.js

```diff
@@ -276,40 +333,50 @@ exports.generateTiddlerFilepath = function(title,options) {
 			}
 		});
 	}
-	// If not, generate a base pathname
-	if(!filepath) {
-		filepath = title;
-		// If the filepath already ends in the extension then remove it
-		if(filepath.substring(filepath.length - extension.length) === extension) {
-			filepath = filepath.substring(0,filepath.length - extension.length);
-		}
-		// Remove any forward or backward slashes so we don't create directories
-		filepath = filepath.replace(/\/|\\/g,"_");
-	}
-	// Don't let the filename start with a dot because such files are invisible on *nix
-	filepath = filepath.replace(/^\./g,"_");
-	// Remove any characters that can't be used in cross-platform filenames
-	filepath = $tw.utils.transliterate(filepath.replace(/<|>|\:|\"|\||\?|\*|\^/g,"_"));
-	// Truncate the filename if it is too long
-	if(filepath.length > 200) {
-		filepath = filepath.substr(0,200);
-	}
-	// If the resulting filename is blank (eg because the title is just punctuation characters)
-	if(!filepath) {
-		// ...then just use the character codes of the title
-		filepath = "";	
-		$tw.utils.each(title.split(""),function(char) {
-			if(filepath) {
-				filepath += "-";
+	// Check for an "isEditableFile" basepath if no pathFilter matches
+	isEditableFile = options.fileInfo ? !!options.fileInfo.basepath : false;
+	if(!filepath && isEditableFile) {
+		var parsed = path.parse(options.fileInfo.basepath);
```

Comment:
I'll take a thorough look in the morning but two quick questions @joshuafontany 
- are we ignoring options.extension if no pathFilter matched?
- for all tiddlers where no pathFilter matched, could we not re-use options.fileInfo.filePath if it exists? This would mean that all tiddlers that do not match a filePath filter get saved back to their original location if there was one.

Reply:
- The file extension is determined via a $utils method call and then passed to generateTiddlerFileInfo. This allows them to be customized completely independently.

 - This depends on your intent for `isEditableFile`, especially in corner-cases where a tiddler starts as one of these tiddlers, gets changed and re-saved to disk, and then gets changed again multiple times until it fails to trigger any FileSystemPath tiddler. Should that one end up in `./tiddlers` or should it "remember" it was loaded via tiddlywiki.files as editable and end up back in the basepath? What if you "accidentally" delete a tiddler and then recreate it (via cloning a similar one)?
    - When a tiddler is deleted, I clear it from `$tw.syncer.tiddlerInfo[title]` but I do not clear it from $tw.boot.files[title] as I was getting some weirness in the callback chain when I tried where I thought it would go. But this also gives us the ability to restore that data if a tiddler with that name is created again while the server is running.
  - Some of my tests had image tiddlers hopping around disk, being saved as .json tiddlers (bad fieds), then as *.txt, etc. I also wanted to run the `isEditableFile` tiddlers through similar series of tests.

Thanks for your notes!